File: CS/CustomFunctionCriteriaOperator.Web/WebApplication.cs
using DevExpress.ExpressApp.Xpo;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.ExpressApp.Web;

using CustomFunctionCriteriaOperator.Module;
using DevExpress.ExpressApp;
using CustomFunctionCriteriaOperator.Module.BusinessObjects;
using DevExpress.Data.Filtering;
using DevExpress.ExpressApp.Security;

namespace CustomFunctionCriteriaOperator.Web {
   public partial class CustomFunctionCriteriaOperatorAspNetApplication : WebApplication {
        protected override void CreateDefaultObjectSpaceProvider(CreateCustomObjectSpaceProviderEventArgs args) {
            args.ObjectSpaceProvider = new XPObjectSpaceProvider(args.ConnectionString, args.Connection);
        }
      private DevExpress.ExpressApp.SystemModule.SystemModule module1;
      private DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule module2;
      private CustomFunctionCriteriaOperator.Module.CustomFunctionCriteriaOperatorModule module3;
      private DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule module6;
      private System.Data.SqlClient.SqlConnection sqlConnection1;
      private SecurityModule securityModule1;
      private SecurityStrategyComplex securityStrategyComplex1;
      private AuthenticationStandard authenticationStandard1;
      private DevExpress.ExpressApp.Validation.ValidationModule module5;

      public CustomFunctionCriteriaOperatorAspNetApplication() {
         InitializeComponent();
      }

      private void CustomFunctionCriteriaOperatorAspNetApplication_DatabaseVersionMismatch(object sender, DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs e) {
            e.Updater.Update();
            e.Handled = true;
      }

      private void InitializeComponent() {
          this.module1 = new DevExpress.ExpressApp.SystemModule.SystemModule();
          this.module2 = new DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule();
          this.module3 = new CustomFunctionCriteriaOperator.Module.CustomFunctionCriteriaOperatorModule();
          this.module5 = new DevExpress.ExpressApp.Validation.ValidationModule();
          this.module6 = new DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule();
          this.sqlConnection1 = new System.Data.SqlClient.SqlConnection();
          this.securityModule1 = new DevExpress.ExpressApp.Security.SecurityModule();
          this.securityStrategyComplex1 = new DevExpress.ExpressApp.Security.SecurityStrategyComplex();
          this.authenticationStandard1 = new DevExpress.ExpressApp.Security.AuthenticationStandard();
          ((System.ComponentModel.ISupportInitialize)(this)).BeginInit();
          // 
          // module5
          // 
          this.module5.AllowValidationDetailsAccess = true;
          // 
          // sqlConnection1
          // 
          this.sqlConnection1.ConnectionString = "Data Source=(local);Initial Catalog=CustomFunctionCriteriaOperator;Integrated Sec" +
              "urity=SSPI;Pooling=false";
          this.sqlConnection1.FireInfoMessageEventOnUserErrors = false;
          // 
          // securityStrategyComplex1
          // 
          this.securityStrategyComplex1.Authentication = this.authenticationStandard1;
          this.securityStrategyComplex1.RoleType = typeof(DevExpress.ExpressApp.Security.Strategy.SecuritySystemRole);
          this.securityStrategyComplex1.UserType = typeof(DevExpress.ExpressApp.Security.Strategy.SecuritySystemUser);
          // 
          // authenticationStandard1
          // 
          this.authenticationStandard1.LogonParametersType = typeof(DevExpress.ExpressApp.Security.AuthenticationStandardLogonParameters);
          // 
          // CustomFunctionCriteriaOperatorAspNetApplication
          // 
          this.ApplicationName = "CustomFunctionCriteriaOperator";
          this.Connection = this.sqlConnection1;
          this.Modules.Add(this.module1);
          this.Modules.Add(this.module2);
          this.Modules.Add(this.module6);
          this.Modules.Add(this.securityModule1);
          this.Modules.Add(this.module3);
          this.Modules.Add(this.module5);
          this.Security = this.securityStrategyComplex1;
          this.LastLogonParametersRead += new System.EventHandler<DevExpress.ExpressApp.LastLogonParametersReadEventArgs>(this.CustomFunctionCriteriaOperatorAspNetApplication_LastLogonParametersRead);
          this.DatabaseVersionMismatch += new System.EventHandler<DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs>(this.CustomFunctionCriteriaOperatorAspNetApplication_DatabaseVersionMismatch);
          ((System.ComponentModel.ISupportInitialize)(this)).EndInit();

      }

      private void CustomFunctionCriteriaOperatorAspNetApplication_LastLogonParametersRead(object sender, LastLogonParametersReadEventArgs e) {
         // Just to read demo user name for logon.
         AuthenticationStandardLogonParameters logonParameters = e.LogonObject as AuthenticationStandardLogonParameters;
         if (logonParameters != null) {
            if (String.IsNullOrEmpty(logonParameters.UserName)) {
               logonParameters.UserName = "Sam";
            }
         }
      }
   }
}

File: VB/CustomFunctionCriteriaOperator.Web/WebApplication.vb
Imports DevExpress.ExpressApp.Xpo
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.ExpressApp.Web

Imports CustomFunctionCriteriaOperator.Module
Imports DevExpress.ExpressApp
Imports CustomFunctionCriteriaOperator.Module.BusinessObjects
Imports DevExpress.Data.Filtering
Imports DevExpress.ExpressApp.Security

Namespace CustomFunctionCriteriaOperator.Web
   Partial Public Class CustomFunctionCriteriaOperatorAspNetApplication
       Inherits WebApplication

        Protected Overrides Sub CreateDefaultObjectSpaceProvider(ByVal args As CreateCustomObjectSpaceProviderEventArgs)
            args.ObjectSpaceProvider = New XPObjectSpaceProvider(args.ConnectionString, args.Connection)
        End Sub
      Private module1 As DevExpress.ExpressApp.SystemModule.SystemModule
      Private module2 As DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule
      Private module3 As CustomFunctionCriteriaOperator.Module.CustomFunctionCriteriaOperatorModule
      Private module6 As DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule
      Private sqlConnection1 As System.Data.SqlClient.SqlConnection
      Private securityModule1 As SecurityModule
      Private securityStrategyComplex1 As SecurityStrategyComplex
      Private authenticationStandard1 As AuthenticationStandard
      Private module5 As DevExpress.ExpressApp.Validation.ValidationModule

      Public Sub New()
         InitializeComponent()
      End Sub

      Private Sub CustomFunctionCriteriaOperatorAspNetApplication_DatabaseVersionMismatch(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs) Handles MyBase.DatabaseVersionMismatch
            e.Updater.Update()
            e.Handled = True
      End Sub

      Private Sub InitializeComponent()
          Me.module1 = New DevExpress.ExpressApp.SystemModule.SystemModule()
          Me.module2 = New DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule()
          Me.module3 = New CustomFunctionCriteriaOperator.Module.CustomFunctionCriteriaOperatorModule()
          Me.module5 = New DevExpress.ExpressApp.Validation.ValidationModule()
          Me.module6 = New DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule()
          Me.sqlConnection1 = New System.Data.SqlClient.SqlConnection()
          Me.securityModule1 = New DevExpress.ExpressApp.Security.SecurityModule()
          Me.securityStrategyComplex1 = New DevExpress.ExpressApp.Security.SecurityStrategyComplex()
          Me.authenticationStandard1 = New DevExpress.ExpressApp.Security.AuthenticationStandard()
          DirectCast(Me, System.ComponentModel.ISupportInitialize).BeginInit()
          ' 
          ' module5
          ' 
          Me.module5.AllowValidationDetailsAccess = True
          ' 
          ' sqlConnection1
          ' 
          Me.sqlConnection1.ConnectionString = "Data Source=(local);Initial Catalog=CustomFunctionCriteriaOperator;Integrated Sec" & "urity=SSPI;Pooling=false"
          Me.sqlConnection1.FireInfoMessageEventOnUserErrors = False
          ' 
          ' securityStrategyComplex1
          ' 
          Me.securityStrategyComplex1.Authentication = Me.authenticationStandard1
          Me.securityStrategyComplex1.RoleType = GetType(DevExpress.ExpressApp.Security.Strategy.SecuritySystemRole)
          Me.securityStrategyComplex1.UserType = GetType(DevExpress.ExpressApp.Security.Strategy.SecuritySystemUser)
          ' 
          ' authenticationStandard1
          ' 
          Me.authenticationStandard1.LogonParametersType = GetType(DevExpress.ExpressApp.Security.AuthenticationStandardLogonParameters)
          ' 
          ' CustomFunctionCriteriaOperatorAspNetApplication
          ' 
          Me.ApplicationName = "CustomFunctionCriteriaOperator"
          Me.Connection = Me.sqlConnection1
          Me.Modules.Add(Me.module1)
          Me.Modules.Add(Me.module2)
          Me.Modules.Add(Me.module6)
          Me.Modules.Add(Me.securityModule1)
          Me.Modules.Add(Me.module3)
          Me.Modules.Add(Me.module5)
          Me.Security = Me.securityStrategyComplex1
          DirectCast(Me, System.ComponentModel.ISupportInitialize).EndInit()

      End Sub

      Private Sub CustomFunctionCriteriaOperatorAspNetApplication_LastLogonParametersRead(ByVal sender As Object, ByVal e As LastLogonParametersReadEventArgs) Handles MyBase.LastLogonParametersRead
         ' Just to read demo user name for logon.
         Dim logonParameters As AuthenticationStandardLogonParameters = TryCast(e.LogonObject, AuthenticationStandardLogonParameters)
         If logonParameters IsNot Nothing Then
            If String.IsNullOrEmpty(logonParameters.UserName) Then
               logonParameters.UserName = "Sam"
            End If
         End If
      End Sub
   End Class
End Namespace
